src/wp-includes/link-template.php

```diff
@@ -1474,6 +1474,8 @@ function get_edit_post_link( $post = 0, $context = 'display' ) {
 	if ( 'wp_template' === $post->post_type || 'wp_template_part' === $post->post_type ) {
 		$slug = urlencode( get_stylesheet() . '//' . $post->post_name );
 		$link = admin_url( sprintf( $post_type_object->_edit_link, $post->post_type, $slug ) );
+	} elseif ( 'wp_navigation' === $post->post_type ) {
```

Comment:
I think we could add an annotation in the `get_edit_post_link()` PHP doc, e.g., something like

`* @since 6.3.0 Adds custom link for wp_navigation post types.`